### PR TITLE
tls: Follow redirects if URL for strict cluster is non-HTTPS

### PR DESCRIPTION
**Pull request by mhrabovcin**
Fixes https://jira.mesosphere.com/browse/DCOS-17992

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/http/CustomRedirectStrategy.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.apache.http.annotation.ThreadingBehavior;
+
+@Contract(
+        threading = ThreadingBehavior.IMMUTABLE
+)
+/**
+ * A copy of {@link org.apache.http.impl.client.LaxRedirectStrategy} with addition support for PUT method redirect
+ * that is necessary for SecretsClient.
+ */
+public class CustomRedirectStrategy extends DefaultRedirectStrategy {
```
Comment (elezar) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/http/CustomRedirectStrategy.java]: Could we extend `LaxRedirectStrategy` here and then first check whether `method=="PUT"` before delegating to `LaxRedirectStrategy.isRedirectable()`?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/auth/ServiceAccountIAMTokenProvider.java [hunk 2]
```diff
@@ -135,6 +136,8 @@ public Executor buildExecutor() throws NoSuchAlgorithmException {
                 httpClientBuilder.setDefaultConnectionTimeout(connectionTimeout);
             }
 
+            httpClientBuilder.setRedirectStrategy(new LaxRedirectStrategy());
```
Comment (elezar) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/auth/ServiceAccountIAMTokenProvider.java]: Is it OK to use the `LaxRedirectStrategy` here?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/http/CustomRedirectStrategy.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    private static final String[] REDIRECT_METHODS = new String[]{"GET", "POST", "PUT", "HEAD", "DELETE"};
+
+    public CustomRedirectStrategy() {
+    }
+
+    protected boolean isRedirectable(String method) {
+        String[] arr$ = REDIRECT_METHODS;
+        int len$ = arr$.length;
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/http/CustomRedirectStrategy.java]: Nitpick: The `$` look funny to me.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/http/CustomRedirectStrategy.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * that is necessary for SecretsClient.
+ */
+public class CustomRedirectStrategy extends DefaultRedirectStrategy {
+    private static final String[] REDIRECT_METHODS = new String[]{"GET", "POST", "PUT", "HEAD", "DELETE"};
+
+    public CustomRedirectStrategy() {
+    }
+
+    protected boolean isRedirectable(String method) {
+        String[] arr$ = REDIRECT_METHODS;
```
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/http/CustomRedirectStrategy.java]: Nitpick: The `$` look funny to me.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/auth/ServiceAccountIAMTokenProvider.java (hunk 2 above)
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/auth/ServiceAccountIAMTokenProvider.java]: Looks so, there's only a `POST` here AFAICT.

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/http/CustomRedirectStrategy.java (hunk 1 above)
Comment (benclarkwood) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/http/CustomRedirectStrategy.java]: That would keep things simpler.

@mhrabovcin can you make that change and add a unit test for the isRedirectable method?

**Pull request merged**